core-errors/src/lib.rs

```diff
@@ -177,20 +184,34 @@ pub enum MemoryError {
     /// The error occurs in attempt to access memory outside wasm program memory.
     #[display(fmt = "Trying to access memory outside wasm program memory")]
     AccessOutOfBounds,
+}
 
-    /// Memory size cannot be zero after grow is applied for memory
-    #[display(fmt = "Memory unexpectedly has zero size after grow")]
-    MemSizeIsZeroAfterGrow,
-
-    /// Memory size cannot be zero after grow is applied for memory
-    // TODO: (issue #1956) make separate error for alloc in allocations context
-    #[display(fmt = "Allocated memory pages or memory size are incorrect")]
-    IncorrectAllocationsSetOrMemSize,
+/// Reservation error.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Hash, derive_more::Display)]
+#[cfg_attr(feature = "codec", derive(Encode, Decode, TypeInfo))]
+#[non_exhaustive]
+pub enum ReservationError {
+    /// An error occurs in attempt to unreserve gas with non-existing reservation ID.
+    #[display(fmt = "Invalid reservation ID")]
+    InvalidReservationId,
+    /// An error occurs in attempt to reserve more gas than available.
+    #[display(fmt = "Insufficient gas for reservation")]
+    InsufficientGasForReservation,
+    /// An error occurs in attempt to reserve more times than allowed.
+    #[display(fmt = "Reservation limit has reached")]
+    ReservationsLimitReached,
+    /// An error occurs in attempt to create reservation for 0 blocks.
+    #[display(fmt = "Reservation duration cannot be zero")]
+    ZeroReservationDuration,
+    /// An error occurs in attempt to reserve zero gas.
+    #[display(fmt = "Reservation amount cannot be zero")]
+    ZeroReservationAmount,
 }
 
 /// Execution error.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, derive_more::Display)]
 #[cfg_attr(feature = "codec", derive(Encode, Decode, TypeInfo))]
+#[non_exhaustive]
 pub enum ExecutionError {
     // TODO: remove some errors in #2199
```

Comment:
Why you don't fix this one ? It will be fixed in right next PR. I think it's very important

Reply:
// TODO: remove some errors in #2199